dictionary/lib/dictionary/word_list.ex

```diff
@@ -1,10 +1,16 @@
 defmodule Dictionary.WordList do
 
+  @name :word_agent
+
+  def start_link() do
+    Agent.start_link(fn -> word_list() end, name: @name)
+  end
+
   def random_word() do
-    word_list()
+    Agent.get(@name, fn state -> state end)
```

Comment:
# -4

What you're doing here is fetching the whole word list from the agent and then selecting a word. It would be better to have

~~~ elixir
Agent.get(@name, &Enum.random//1)
~~~